028-service-binding.md

```diff
@@ -0,0 +1,276 @@
+    ssl.truststore.password: # Password for protecting the Strimzi cluster CA certificate PKCS #12 archive file (copied from ca.password in <clustername>-cluster-ca-cert secret)
+    # Provided if selected user is SCRAM auth:
+    username: # SCRAM username
+    password: # SCRAM password
+    sasl.jaas.config: # sasl jaas config string for use by Java applications
+    sasl.mechanism: SCRAM-SHA-512
+    # Provided if selected user is mTLS:
+    ssl.keystore.p12: # client certificate for the consuming client PKCS #12 archive file for storing certificates and keys (copied from user.p12 in KafkaUser secret)
+    ssl.keystore.password: # password for protecting the client certificate PKCS #12 archive file (copied from user.password in KafkaUser secret)
+    ssl.keystore.crt: # certificate for the consuming client signed by the clients' CA (copied from user.crt in KafkaUser secret)
+    ssl.keystore.key: # private key for the consuming client (copied from user.key in KafkaUser secret)
+```
+
+Naming of secret keys:
+
+This proposal previously considered choosing secret keys that could be reused by other (non-Kafka) operators that use mutualTLS or SASL SCRAM authentication. However, given 
+that Kafka applications generally have a well known list of configuration options they provide, the proposal instead uses those well-known keys. This will make it clearer 
+to application developers what the different keys should be used for.
+
+### Implementation
+
+This proposal will introduce a new `KafkaAccess` operator. The operator will be separate from the existing `strimzi-kafka-operator` and live in its own GitHub repo in the `strimzi` organization.
+
+The actions taken during the reconcile loop of the new operator will be:
+1. The operator can be deployed at cluster-scope where it will watch resources across all namespaces, or in a single namespace, where it will only watch resources in that namespace.
+2. On start-up the operator will watch all resources of Kind `KafkaAccess`.
+3. For any `KafkaAccess` resource present it will add a status of NotReady to that CR.
+4. For each `KafkaAccess` resource it will perform a GET request on the Kafka cluster that is 
+referenced in the `KafkaAccess` and pick out the configured listeners.
+   1. If this or any subsequent GET requests fail due to a 401 (unauthorised) error, the status reason will be updated to reflect this.
+5. The operator will determine which listener to choose:
+   1. If there are no listeners in the `Kafka` CR the reconcile loop will fail.
+   2. If the `KafkaAccess` specifies a listener, the operator will choose this listener.
+   3. If there are no listeners specified in the `KafkaAccess`, but only one listener in the `Kafka` CR, the operator will choose that listener.
+   4. If there are no listeners specified in the `KafkaAccess`, but multiple listeners listed in the `Kafka` CR, the operator 
```

Comment:
Why does the `tls` property matter here? The `KafkaAcess` does not seem to touch it.

Reply:
The Kafka Access operator needs to know if the listener has `tls` true or not, because if it is true it should go and fetch the cluster ca certificate and put that in the secret.